src/process_utility.c

```diff
@@ -834,13 +924,18 @@ process_rename_table(Cache *hcache, Oid relid, RenameStmt *stmt)
 
 		if (NULL != chunk)
 			ts_chunk_set_name(chunk, stmt->newname);
+
+		return false;
 	}
-	else
-		ts_hypertable_set_name(ht, stmt->newname);
+
+	ts_hypertable_set_name(ht, stmt->newname);
```

Comment:
I think we can just move this to `process_ddl_command_end` and avoid the special logic here. But I'll leave it up to you. You can decide.